src/DynamoCoreWpf/ViewModels/Core/StateMachine.cs

```diff
@@ -525,8 +525,12 @@ internal bool HandleLeftButtonDown(object sender, MouseButtonEventArgs e)
                             // a code block node being created, in which case we
                             // should keep the input focus on the code block to 
                             // avoid it being dismissed (with empty content).
-                            // 
-                            CreateCodeBlockNode(mouseDownPos);
+                            //
+                            // If Shift is pressed, CBN shouldn't be created.
+                            // Shift Modifier indicates, that user tries to call InCanvasSearch
+                            // by using Shift + DoubleClick.
+                            if (Keyboard.Modifiers != ModifierKeys.Shift)
+                                CreateCodeBlockNode(mouseDownPos);
```

Comment:
There is no need to return focus to search. As with CBN, focus will go to InCanvasSearch search text box. Also I don't see any other places, where we should proceed Shift... for now.